### PR TITLE
Fix any type issue for the functions in aggregation.ts Addresses Issue#<123> and MapNetworkNodes

### PR DESCRIPTION
**Pull request by mozartfish**
This PR fixes the issues with the any type in aggregation.ts and makes it easier to know what the types of objects are when using these specific functions for aggregation and adds a new function for keeping track of nodes and ids in a network

### CODE REVIEW

File: src/lib/aggregation.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export function superGraph(nodes: any[], edges: any[], attribute: string) {
 
     // remove the properties that will not be used
     // and properties that will be recalculated for visualization
-    delete newNode.neighbors;
+    newNode.neighbors = [];
```
Comment (waxlamp) [inline src/lib/aggregation.ts:18]: Why this change?

File: src/lib/aggregation.ts [hunk 2]
```diff
@@ -112,3 +112,13 @@ export function superGraph(nodes: any[], edges: any[], attribute: string) {
...
+
+// function that maps node names to nodes
+export function MapNetworkNodes(nodes: Node[]) {
+  const nodeMap = new Map<string, Node>();
+  nodes.forEach((node: Node) => {
+    nodeMap.set(node.id, node);
+  });
+
+  return nodeMap;
+}
```
Comment (waxlamp) [inline src/lib/aggregation.ts:124]: This doesn't seem like it belongs in this PR.

**Comment by waxlamp** [comment]
@mozartfish, can you edit the PR description and include the phrase `Closes #<issue number>` so that the appropriate issue or issues are closed when this is merged?

File: src/lib/aggregation.ts (hunk 1 above)
Comment (mozartfish) [reply src/lib/aggregation.ts:18]: [
Screenshot from 2020-12-04 11-06-44
]

Comment (mozartfish) [reply src/lib/aggregation.ts:18]: there was so many comments about this issue on github so I tried some stuff and got this error to go away when I set the neighbors to empty

Comment (mozartfish) [reply src/lib/aggregation.ts:18]: according to this post https://stackoverflow.com/questions/63702057/what-is-the-logic-behind-error-the-operand-of-a-delete-operator-must-be-opti, when changing the  type of nodes from any to node, I violated the interface contract for node since it requires a field called neighbors. What I did with this change to maintain the contract of the interface is set an empty list of neighbors so that the neighbors field would not be undefined and there would not be a contact violation.

Comment (mozartfish) [reply src/lib/aggregation.ts:18]: https://stackoverflow.com/questions/63702057/what-is-the-logic-behind-error-the-operand-of-a-delete-operator-must-be-opti
